src/state/trajectory/logics.ts

```diff
@@ -575,6 +578,35 @@ const convertFileLogic = createLogic({
     type: CONVERT_FILE,
 });
 
+const receiveConvertedFileLogic = createLogic({
+    process(deps: ReduxLogicDeps, dispatch, done) {
+        const { action, getState } = deps;
+        const currentState = getState();
+        const conversionStatus = getConversionStatus(currentState);
+        const simulariumController = getSimulariumController(currentState);
+        const simulariumFile = action.payload;
+
+        simulariumController.changeFile(
+            netConnectionSettings,
+            simulariumFile.name,
+            true
+        );
```

Comment:
calling changeFile with this boolean flag ensure we use the same websocket connection from the conversion process

Reply:
is changeFile an async function? how do you know it was successful/catch errors if it failed?

Reply:
yes `changeFile` returns a Promise

Reply:
then you should either await it's return or have a `then` block that encompasses everything you want to do if it's successful. and have a fail state if it doesn't work